Models/USBSpectrometer.cs

```diff
@@ -0,0 +1,696 @@
+        public virtual bool isInGaAs => (featureIdentification.boardType == BOARD_TYPES.INGAAS_FX2 || eeprom.detectorName.StartsWith("g", StringComparison.CurrentCultureIgnoreCase));
+
+        static UsbDeviceConnection udc;
+        static UsbDevice acc;
+
+        static public USBSpectrometer getInstance()
+        {
+            if (instance is null)
+                instance = new USBSpectrometer();
+            return instance;
+        }
+
+        public static void setInstance(USBSpectrometer inst)
+        {
+            instance = inst;
+        }
+
+        public USBSpectrometer()
+        {
+            reset();
+        }
+
+        public USBSpectrometer(UsbDeviceConnection udcI, UsbDevice accI) : base()
+        {
+            reset();
+            logger.debug("connecting to usb device vid: {0:x4}, pid: {1:x4}", accI.VendorId, accI.ProductId);
+            featureIdentification = new FeatureIdentification(accI.VendorId, accI.ProductId);
+            battery = new Battery();
+
+            udc = udcI;
+            acc = accI;
+            connect();
+            //instance = new USBSpectrometer(udc, acc);
+            //instance.connect();
+        }
+
+        public override void disconnect()
+        {
+            udc.ReleaseInterface(acc.GetInterface(0));
+            udc.Close();
+            isConnected = false;
+            logger.info("closed usb device");
+        }
+
+        public async Task<bool> initAsync()
+        {
+            connect();
+            var pages = await readEEPROMAsync();
+            if (pages is null)
+            {
+                logger.error("Spectrometer.initAsync: failed to read EEPROM");
+                return false;
+            }
+
+            logger.debug("Spectrometer.initAsync: parsing EEPROM");
+            if (!eeprom.parse(pages))
+            {
+                logger.error("Spectrometer.initAsync: failed to parse EEPROM");
+                return false;
+            }
+
+            ////////////////////////////////////////////////////////////////////
+            // post-process EEPROM
+            ////////////////////////////////////////////////////////////////////
+
+            logger.debug("Spectrometer.initAsync: post-processing EEPROM");
+            pixels = eeprom.activePixelsHoriz;
+            laserExcitationNM = eeprom.laserExcitationWavelengthNMFloat;
+
+            logger.debug("Spectrometer.initAsync: computing wavecal");
+            wavelengths = Util.generateWavelengths(pixels, eeprom.wavecalCoeffs);
+
+            if (laserExcitationNM > 0)
+                wavenumbers = Util.wavelengthsToWavenumbers(laserExcitationNM, wavelengths);
+            else
+                wavenumbers = null;
+
+            logger.debug("Spectrometer.initAsync: generating pixel axis");
+            generatePixelAxis();
+
+            // set this early so battery and other BLE calls can progress
+            paired = true;
```

Comment:
Are we using "paired" to indicate "enumerated"? How does this differ from isConnected?

Reply:
I added paired to the base class but not isConnected 😆, yeah these are redundant